src/realm/deppart/byfield_gpu_impl.hpp

```diff
@@ -0,0 +1,261 @@
+#pragma once
+#include "realm/deppart/byfield.h"
+#include "realm/deppart/byfield_gpu_kernels.hpp"
+#include "realm/deppart/partitions_gpu_impl.hpp"
+#include <cub/cub.cuh>
+#include "realm/nvtx.h"
+
+namespace Realm {
+
+template <int N, typename T>
+struct RectFieldVolumeOp {
+  __device__ __forceinline__
+  size_t operator()(const Rect<N,T>& r) const {
+    return r.volume();
+  }
+};
+
+// everything in header form:
+template <int N, typename T, typename FT>
+void GPUByFieldMicroOp<N,T,FT>::gpu_populate_bitmasks()
+{
+
+    nvtx_range_push("cuda", "byfield_gpu_populate_bitmasks");
+
+    cudaStream_t stream;
+    CUDA_CHECK(cudaStreamCreate(&stream), stream);
+
+    Memory my_mem = field_data[0].inst.get_location();
```

Comment:
Are all the instances gauranteed to be in the same memory? Do you check this anywhere?